javascript/README.md

```diff
@@ -283,6 +294,125 @@ const { a, ...noA } = copy; // noA => { b: 2, c: 3 }
 #### Resources
 
 - ESLint: [prefer-object-spread]
-- JSPerf: [Shallow Copy Objects][JSPerf: Shallow Copy Objects]
+- JSPerf: [Shallow Copy Objects]
+
+[⇧ top](#javascript-guide)
+
+---
+
+## Arrays
+
+### 3.1 Array Creation
+
+Use the literal syntax for array creation.
+
+> Why? Use of the `Array` constructor to construct a new array is generally discouraged in favor of array literal notation because of the single-argument pitfall and because the `Array` global may be redefined.
+
+#### Examples
+
+🚫 Nope. 🚫
+
+```js
+const items = new Array();
+```
+
+🎉 Yep! 🎉
+
+```js
+const items = [];
+```
+
+#### Resources
+
+- ESLint: [no-array-constructor]
+
+
+### 3.2 Adding Items To Arrays
+
+Use [Array.prototype.push()][Array.prototype.push] instead of direct assignment to add items to an array.
+
+#### Examples
+
+🚫 Nope. 🚫
+
+```js
+const someStack = [];
+someStack[someStack.length] = 'abracadabra';
+```
+
+🎉 Yep! 🎉
+
+```js
+const someStack = [];
+someStack.push('abracadabra');
+```
+
+#### Resources
+
+- JSPerf: [Adding Array Items]
+
+### 3.3 Array Shallow-Copy
+
+Use [array spread syntax][Array Literal Spread Syntax] `...` to shallow-copy arrays.
+
+> Why? Better overall performance.
+
+#### Examples
+
+🚫 Nope. 🚫
+
+```js
+// Too slow
+const animals = ['ant', 'bison', 'camel', 'duck', 'elephant'];
+const len = animals.length;
+const animalsCopy = [];
+let i;
+
+for (i = 0; i < len; i ++) {
+  animalsCopy[i] = animals[i];
+}
+
+// Works but is not preferred
+const animals = ['ant', 'bison', 'camel', 'duck', 'elephant'];
+const animalsCopy = animals.slice();
+```
+
+🎉 Yep! 🎉
+
+```js
+const animals = ['ant', 'bison', 'camel', 'duck', 'elephant'];
+const animalsCopy = [...animals];
+```
+
+#### Resources:
+
+- JSPerf: [Shallow Copy Arrays]
+
+### 3.4 Arrays From Iterables
+
+To convert an iterable object (e.g. [NodeList]) to an array, use [array spread syntax][Array Literal Spread Syntax] `...` instead of [Array.from].
+
+> Why? Better performance.
+
+#### Examples
+
+🚫 Nope. 🚫
+
+```js
+const paragraphs = document.querySelectorAll('p');
+const nodes = Array.from(paragraphs);
+```
+
+🎉 Yep! 🎉
+
+```js
+const paragraphs = document.querySelectorAll('p');
+const nodes = [...paragraphs];
+```
+
+#### Resources
```

Comment:
https://github.com/sindresorhus/eslint-plugin-unicorn/blob/master/docs/rules/prefer-spread.md

Reply:
Thanks @calebeby! Added this URL. 😉